# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeOperation.h
# Commit: f9cf0740eef2
# Full Hash: f9cf0740eef2fcae3e2f93afe49df1569bd89c03
# Author: Adam Vandolder <avandolder@mozilla.com>
# Date: 2024-05-15 20:24:18
# Regressor Bug: 1890428
# File Overlap Count: 1
# Description:
#   Bug 1890428 - Add serialization support for Declarative Shadow DOM. r=webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D206955
# ==============================================================================

diff -r 37e2a10a9022 -r f9cf0740eef2 parser/html/nsHtml5TreeOperation.h
--- a/parser/html/nsHtml5TreeOperation.h	Tue May 14 16:43:27 2024 +0000
+++ b/parser/html/nsHtml5TreeOperation.h	Tue May 14 16:43:28 2024 +0000
@@ -282,6 +282,7 @@
   nsIContent** mTemplateNode;
   mozilla::dom::ShadowRootMode mShadowRootMode;
   bool mShadowRootIsClonable;
+  bool mShadowRootIsSerializable;
   bool mShadowRootDelegatesFocus;
 
   explicit opGetShadowRootFromHost(nsIContentHandle* aHost,
@@ -289,12 +290,14 @@
                                    nsIContentHandle* aTemplateNode,
                                    mozilla::dom::ShadowRootMode aShadowRootMode,
                                    bool aShadowRootIsClonable,
+                                   bool aShadowRootIsSerializable,
                                    bool aShadowRootDelegatesFocus) {
     mHost = static_cast<nsIContent**>(aHost);
     mFragHandle = static_cast<nsIContent**>(aFragHandle);
     mTemplateNode = static_cast<nsIContent**>(aTemplateNode);
     mShadowRootMode = aShadowRootMode;
     mShadowRootIsClonable = aShadowRootIsClonable;
+    mShadowRootIsSerializable = aShadowRootIsSerializable;
     mShadowRootDelegatesFocus = aShadowRootDelegatesFocus;
   }
 };